domain-models-runtime/src/main/java/org/folio/rest/tools/utils/TenantLoading.java

```diff
@@ -202,121 +202,144 @@ private static void endWithXHeaders(HttpClientRequest req, Map<String, String> h
   private static String getId(LoadingEntry loadingEntry, URL url, String content,
     Future<Void> f) {
 
-    String id = null;
-    switch (loadingEntry.strategy) {
-      case BASENAME:
-        int base = url.getPath().lastIndexOf(File.separator);
-        int suf = url.getPath().lastIndexOf('.');
-        if (base == -1) {
-          f.handle(Future.failedFuture("No basename for " + url.toString()));
-          return null;
-        }
-        if (suf > base) {
-          id = url.getPath().substring(base, suf);
-        } else {
-          id = url.getPath().substring(base);
-        }
-        break;
-      case CONTENT:
-        JsonObject jsonObject = new JsonObject(content);
-        id = jsonObject.getString(loadingEntry.idProperty);
-        if (id == null) {
-          log.warn("Missing property "
-            + loadingEntry.idProperty + " for url=" + url.toString());
-
-          f.handle(Future.failedFuture("Missing property "
-            + loadingEntry.idProperty + " for url=" + url.toString()));
-          return null;
-        }
-        try {
-          id = URLEncoder.encode(id, StandardCharsets.UTF_8.name());
-        } catch (UnsupportedEncodingException ex) {
-          f.handle(Future.failedFuture("Encoding of " + id + FAILED_STR));
-          return null;
-        }
-        break;
-      case RAW_PUT:
-      case RAW_POST:
-        break;
+    try {
+      String id = null;
+      switch (loadingEntry.strategy) {
+        case BASENAME:
+          int base = url.getPath().lastIndexOf(File.separator);
+          int suf = url.getPath().lastIndexOf('.');
+          if (base == -1) {
+            f.handle(Future.failedFuture("No basename for " + url.toString()));
+            return null;
+          }
+          if (suf > base) {
+            id = url.getPath().substring(base, suf);
+          } else {
+            id = url.getPath().substring(base);
+          }
+          break;
+        case CONTENT:
+          JsonObject jsonObject = new JsonObject(content);
+          id = jsonObject.getString(loadingEntry.idProperty);
+          if (id == null) {
+            log.warn("Missing property "
+              + loadingEntry.idProperty + " for url=" + url.toString());
+
+            f.handle(Future.failedFuture("Missing property "
+              + loadingEntry.idProperty + " for url=" + url.toString()));
+            return null;
+          }
+          try {
+            id = URLEncoder.encode(id, StandardCharsets.UTF_8.name());
+          } catch (UnsupportedEncodingException ex) {
+            f.handle(Future.failedFuture("Encoding of " + id + FAILED_STR));
+            return null;
+          }
+          break;
+        case RAW_PUT:
+        case RAW_POST:
+          break;
+      }
+      return id;
+    } catch (Exception ex) {
+      f.handle(Future.failedFuture("exception: " + ex.getLocalizedMessage()));
```

Comment:
This swallows the stacktrace and does not add information to the message. My suggestion:
```suggestion
      f.handle(Future.failedFuture(ex));
```
Note: ex.getLocalizedMessage() uses the locale of the JVM, this is not the locale of the client; therefore ex.getMessage() should be used instead.